x-pack/plugins/cases/server/services/so_references.ts

```diff
@@ -38,6 +39,36 @@ export const getAttachmentSOExtractor = (attachment: Partial<CommentRequest>) =>
   return new SOReferenceExtractor(fieldsToExtract);
 };
 
+type OptionalAttributes<T> = PartialField<SavedObject<T>, 'attributes'>;
+
+/**
+ * This function should be used when the attributes field could be undefined. Specifically when
+ * performing a bulkGet within the core saved object library. If one of the requested ids does not exist in elasticsearch
+ * then the error field will be set and attributes will be undefined.
+ */
+export const injectAttachmentAttributesAndHandleErrors = (
+  savedObject: OptionalAttributes<CommentAttributesWithoutRefs>,
+  persistableStateAttachmentTypeRegistry: PersistableStateAttachmentTypeRegistry
+): OptionalAttributes<CommentAttributes> => {
+  if (!hasAttributes(savedObject)) {
+    // we don't actually have an attributes field here so the type doesn't matter, this cast is to get the types to stop
+    // complaining though
+    return savedObject as OptionalAttributes<CommentAttributes>;
+  }
+
+  const soExtractor = getAttachmentSOExtractor(savedObject.attributes);
+  const so = soExtractor.populateFieldsFromReferences<CommentAttributes>(savedObject);
+  const injectedAttributes = injectPersistableReferencesToSO(so.attributes, so.references, {
+    persistableStateAttachmentTypeRegistry,
+  });
+
+  return { ...so, attributes: { ...so.attributes, ...injectedAttributes } };
```

Comment:
Should we use `injectAttachmentSOAttributesFromRefs` instead? The code seems to be identical. If types is an issue we can cast it.